### PR TITLE
Add smart-proxy into dependency calculation

### PR DESCRIPTION
**Pull request by ehelms**
At the time of this update here is the list, this shows it would catch the sinatra / mustermann issue:

```
ackages/foreman/rubygem-faraday 1.10.0
packages/foreman/rubygem-google-apis-compute_v1 0.44.0
packages/foreman/rubygem-google-apis-monitoring_v3 0.31.0
packages/foreman/rubygem-representable 3.2.0
packages/foreman/rubygem-rest-client 2.1.0
packages/katello/rubygem-katello 4.6.0.pre.master
packages/katello/rubygem-pulp_ostree_client 2.0.0a7.dev1658981930
packages/foreman/rubygem-gssapi 1.3.1
packages/foreman/rubygem-jwt 2.4.1
packages/foreman/rubygem-mustermann 2.0.2
packages/foreman/rubygem-rack-test 2.0.2
packages/foreman/rubygem-rake-compiler 1.2.0
packages/foreman/rubygem-redfish_client 0.5.4
packages/foreman/rubygem-rubyipmi 0.11.1
packages/foreman/rubygem-server_sent_events 0.1.3
packages/foreman/rubygem-sinatra 2.2.2
packages/foreman/rubygem-unicode-display_width 1.6.1
packages/foreman/rubygem-xmlrpc 0.3.2
packages/foreman/rubygem-clamp 1.3.0
packages/foreman/rubygem-faraday 0.17.5
packages/foreman/rubygem-fast_gettext 1.1.2
packages/foreman/rubygem-gettext 3.2.9
packages/foreman/rubygem-hashie 5.0.0
packages/foreman/rubygem-unicode-display_width 1.6.1
```

### CODE REVIEW

File: list_updatable_packages [hunk 1]
```diff
@@ -162,9 +168,11 @@ def build_dependency_matrix(gems: dict, project: str='foreman') -> Generator[dic
 def main() -> None:
     foreman_dependencies = list(build_dependency_matrix(gems=parse_gemfile_lock('foreman'), project='foreman'))
     katello_dependencies = list(build_dependency_matrix(gems=parse_gemfile_lock('katello'), project='katello'))
+    foreman_proxy_dependencies = list(build_dependency_matrix(gems=parse_gemfile_lock('foreman-proxy'), project='foreman'))
+    foreman_installer_dependencies = list(build_dependency_matrix(gems=parse_gemfile_lock('foreman-installer'), project='foreman'))
     plugins = list(build_plugin_matrix(find_specs()))
 
-    matrix = plugins + foreman_dependencies + katello_dependencies
+    matrix = plugins + foreman_dependencies + katello_dependencies + foreman_proxy_dependencies + foreman_installer_dependencies
```
[the remark this answers is not in the document]
Comment (ehelms) [reply list_updatable_packages:175]: As a start, it could be good to have two versions showing up with PRs for a single project -- should provide a small red flag that something is amiss. Other options would be to fail the updater and hope someone notices or remove from the list of updatables any package that has a dependency conflict.

[the remark this answers is not in the document]
Comment (ekohl) [reply list_updatable_packages:175]: I looked at it and it comes from the installer. For now I'd leave the installer out of it. The only dependencies from it we care about are kafo and puppet-strings.

To update things, I'd have to do a release of librarian-puppet (to get https://github.com/voxpupuli/librarian-puppet/commit/e1c4a261a93f838362d144c2d12f266369b6a13c out) and I don't have time for that right now.

So short term:
```suggestion
    matrix = plugins + foreman_dependencies + katello_dependencies + foreman_proxy_dependencies
```

[the remark this answers is not in the document]
Comment (ehelms) [reply list_updatable_packages:175]: The installer dependencies identify things like clamp that as the output above shows do need an upgrade - which will be true for foreman_maintain, and hammer. Which hammer we could add in as well given it has a source release job.

I was not following the effect librarian-puppet has on these dependency calculations - can you expand?

[the remark this answers is not in the document]
Comment (ekohl) [reply list_updatable_packages:175]: If you look at [`Gemfile.lock`](https://ci.theforeman.org/job/foreman-installer-develop-source-release/lastSuccessfulBuild/artifact/Gemfile.lock/*view*/) from the installer, you see:
```
    puppet_forge (2.3.4)
      faraday (>= 0.9.0, < 0.18.0, != 0.13.1)
      faraday_middleware (>= 0.9.0, < 0.15.0)
      gettext-setup (~> 0.11)
      minitar
      semantic_puppet (~> 1.0)
```

That is where this comes from:
```
    faraday (0.17.5)
      multipart-post (>= 1.2, < 3)
    faraday_middleware (0.14.0)
      faraday (>= 0.7.4, < 1.0)
```

So that ends up pulling in faraday 0.17.5. However, `puppet_forge` is not a dependency we care about in our RPM runtime. It is only a build dependency in our CI.

> The installer dependencies identify things like clamp that as the output above shows do need an upgrade - which will be true for foreman_maintain, and hammer. Which hammer we could add in as well given it has a source release job.

I'm not sure it matters. Hammer can pull in clamp too and we have nightly packages for hammer. The only thing that we care about is [this line](https://github.com/theforeman/foreman-installer/blob/b435f27e9a781d81afe8003cd7852d486770bb4b/Gemfile#L7) to pull in Kafo. We maintain Kafo ourselves so if a new release comes out, we can package it at the same time.

Actually, right now Kafo pulls in clamp with an upper bound of `< 1.3.1`. That's something we should resolve too. https://github.com/mdub/clamp/commit/4b8496d59faf14929f01689b58004e78670b8428 is something we'll need once released.

So in short: I think adding the installer dependency adds guaranteed false positives and gives no benefit.

**Comment by ehelms** [comment]
Updated to just hammer-cli

**Comment by ehelms** [comment]
@ekohl Are you OK with this change now?

**Review by ekohl — APPROVED** [review]
